Source/Core/InputCommon/ControllerInterface/CemuHookUDPServer/CemuHookUDPServer.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <chrono>
+
+#include "Common/Logging/Log.h"
+#include "Common/Matrix.h"
+#include "InputCommon/ControllerInterface/CemuHookUDPServer/CemuHookUDPServerProto.h"
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "SFML/Network/UdpSocket.hpp"
+
+namespace ciface::CemuHookUDPServer
+{
+void Init();
+void PopulateDevices();
+void DeInit();
+void SaveSettings();
+
+class Device : public Core::Device
+{
+private:
+  template <class T>
+  class Button : public Core::Device::Input
+  {
+  public:
+    Button(std::string name, const T& buttons, unsigned mask)
+        : m_name(name), m_buttons(buttons), m_mask(mask)
+    {
+    }
+    std::string GetName() const override { return m_name; }
+    ControlState GetState() const override { return (m_buttons & m_mask) > 0; }
+
+  private:
+    const std::string m_name;
```

Comment:
Was this intended to be a const reference? If not, then this variable can be moved into from within the constructor instead of copying it.